src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -734,16 +741,13 @@ private void printEnclosedStackTrace(PrintStreamOrWriter s,
             int framesInCommon = trace.length - 1 - m;
 
             // Print our stack trace
-            s.println(prefix + caption + this);
-            for (int i = 0; i <= m; i++)
-                s.println(prefix + "\tat " + trace[i]);
-            if (framesInCommon != 0)
-                s.println(prefix + "\t... " + framesInCommon + " more");
+            s.println(prefixCaption.concat(this.toString()));
+            printStackTrace0(s, prefix, trace, m, framesInCommon);
 
             // Print suppressed exceptions, if any
             for (Throwable se : getSuppressed())
                 se.printEnclosedStackTrace(s, trace, SUPPRESSED_CAPTION,
```

Comment:
Shouldn't SUPPRESSED_CAPTION be added a prefix?

I think we should revert the `prefixCaption` argument change; you should add a line `String prefixCaption = prefix.concat(caption);` instead.